s3torchconnector/tst/unit/test_s3_client.py

```diff
@@ -82,3 +84,34 @@ def test_user_agent_always_starts_with_package_version(comments):
     if comments_str:
         assert comments_str in s3_client.user_agent_prefix
         assert comments_str in s3_client._client.user_agent_prefix
+
+
+@given(
+    part_size=integers(min_value=5 * 1024, max_value=5 * 1024 * 1024),
+    throughput_target_gbps=floats(min_value=10.0, max_value=100.0),
+)
+def test_s3_client_custom_config(part_size: int, throughput_target_gbps: float):
+    # Part size must have values between 5MiB and 5GiB
+    part_size = part_size * 1024
+    s3_client = S3Client(
+        region=TEST_REGION,
+        s3client_config=S3ClientConfig(
+            part_size=part_size,
+            throughput_target_gbps=throughput_target_gbps,
+        ),
+    )
+    assert s3_client._client.part_size == part_size
+    assert abs(s3_client._client.throughput_target_gbps - throughput_target_gbps) < 1e-9
```

Comment:
Comment for all of these, why don't we do an equality assertion here?
If that doesn't work, pytest actually has a helper for this: https://stackoverflow.com/a/39623614

Reply:
It should work, but I wanted to emphasize we're comparing floats